src/preprocessing/preprocessing.py

```diff
@@ -0,0 +1,245 @@
+        # columns that would be added later after one-hot encoding each class
+        self.added_features = []
+        self.numerical_data = [
+            "google_places_rating",
+            "google_places_user_ratings_total",
+            "google_places_confidence",
+            "reviews_sentiment_score",
+            "review_avg_grammatical_score",
+            "review_polarization_score",
+            "review_highest_rating_ratio",
+            "review_lowest_rating_ratio",
+            "review_rating_trend",
+        ]
+        # numerical data that need scaling
+        self.data_to_scale = ["google_places_user_ratings_total"]
+
+        # categorical data that needs one-hot encoding
+        self.categorical_data = [
+            # "number_country",
+            # "number_area",
+            "google_places_detailed_type",
+            "review_polarization_type",
+        ]
+
+        self.class_labels = "MerchantSizeByDPV"
+
+    def filter_out_null_data(self):
+        self.preprocessed_df = self.preprocessed_df[
+            self.preprocessed_df["google_places_rating"].notnull()
+        ]
+
+    def fill_missing_values(self, column, strategy="constant"):
+        if column in self.preprocessed_df.columns:
+            imputer = SimpleImputer(strategy=strategy)
+            self.preprocessed_df[column] = imputer.fit_transform(
+                self.preprocessed_df[[column]]
+            )
+        else:
+            log.info(f"The column '{column}' does not exist in the DataFrame.")
+
+        return self.preprocessed_df
+
+    def standard_scaling(self, column):
+        # scales the data in such that the mean of the data becomes 0 and the standard deviation becomes 1.
+        if column in self.preprocessed_df.columns:
+            scaler = StandardScaler()
+            self.preprocessed_df[column] = scaler.fit_transform(
+                self.preprocessed_df[[column]]
+            )
+        return self.preprocessed_df
+
+    def min_max_scaling(self, column):
+        # scales the data to a given range, usually between 0 and 1.
+        if column in self.preprocessed_df.columns:
+            scaler = MinMaxScaler()
+            self.preprocessed_df[column] = scaler.fit_transform(
+                self.preprocessed_df[[column]]
+            )
+        return self.preprocessed_df
+
+    def robust_scaling(self, column):
+        if column in self.preprocessed_df.columns:
+            scaler = RobustScaler()
+            self.preprocessed_df[column] = scaler.fit_transform(
+                self.preprocessed_df[[column]]
+            )
+        return self.preprocessed_df
+
+    def normalization(self, column):
+        if column in self.preprocessed_df.columns:
+            scaler = Normalizer()
+            self.preprocessed_df[column] = scaler.fit_transform(
+                self.preprocessed_df[[column]]
+            )
+        return self.preprocessed_df
+
+    def remove_outliers_zscore(self, column):
+        THRESHOLD = 3

```

Comment:
I assume this corresponds to a 99% interval or is it more of an intuitive value?

Reply:
It is more of a conventional, a rule-of-thumb value, it worked well before.